VidaToDo/Features/Todos/Todo Creation Form/Model/FormViewModel.swift

```diff
@@ -6,39 +6,45 @@
 //  Copyright © 2018 Vida Health. All rights reserved.
 //
 
-// BRICE: Do we like this file structure?
-class FormViewModel {
+struct ToDoFormSheetData {
+    let title: String
+    let due: Date
+    let priortiy: ToDoTask.Priority
+}
 
-    private let disposeBag = DisposeBag()
+// TODO: Do we like this file structure?
+class FormViewModel {
 
-    var isValid: Observable<Bool>?
-    var hasSubmitted: Observable<Bool>? {
+    // Public API
+    public var isValid: Observable<Bool> {
+        return _validData.asObservable().map({ (data) -> Bool in
+            return data != nil
+        })
+    }
+    public var hasSubmitted: Observable<Bool> {
         return _hasSubmitted.asObservable()
     }
 
+    // Subjects
+    public var _validData = Variable<ToDoFormSheetData?>(nil)
     private var _hasSubmitted = Variable<Bool>(false)
 
-    var latestValidData: (String?, Date, ToDoTask.Priority)?
-
-    let manager = TaskToDoManager()
+    // Private
+    private let bag = DisposeBag()
```

Comment:
Looks like this was a deliberate rename? Are you proposing that we name `disposeBag` `bag` from now on?

Reply:
yes. its easy, fast, succinct and fun.

Reply:
I actually vote for `purse`.

Reply:
`backpack`

Reply:
why not 'b' ?

Reply:
Why not `_`